cpu/lm4f120/periph/uart.c

```diff
@@ -31,120 +31,217 @@
+    unsigned long mask_rx;            /**< Pin mask for RX*/
+    unsigned long mask_tx;            /**< Pin mask for TX*/
+} uart_conf_fixed_t;
+
+static const uart_conf_fixed_t ucf[] = {
+    {
+        .uart_sysctl    = SYSCTL_PERIPH_UART0,
+        .uart_base      = UART0_BASE,
+        .uart_irq_chan  = UART0_IRQn,
+        .uart_im_r      = &UART0_IM_R,
+
+        .gpio_sysctl = SYSCTL_PERIPH_GPIOA,
+        .gpio_port   = GPIO_PORTA_BASE,
+        .rx             = GPIO_PA0_U0RX,
+        .tx             = GPIO_PA1_U0TX,
+        .mask_rx        = (GPIO_PIN_0),
+        .mask_tx        = (GPIO_PIN_1),
+    },
+    {
+        .uart_sysctl    = SYSCTL_PERIPH_UART1,
+        .uart_base      = UART1_BASE,
+        .uart_irq_chan  = UART1_IRQn,
+        .uart_im_r      = &UART1_IM_R,
+
+        #ifndef UART1_USE_ALT_PIN
+        .gpio_sysctl = SYSCTL_PERIPH_GPIOB,
+        .gpio_port   = GPIO_PORTB_BASE,
+        .rx             = GPIO_PB0_U1RX,
+        .tx             = GPIO_PB1_U1TX,
+        .mask_rx        = (GPIO_PIN_0),
+        .mask_tx        = (GPIO_PIN_1),
+        #else
+        .gpio_sysctl = SYSCTL_PERIPH_GPIOC,
+        .gpio_port   = GPIO_PORTC_BASE,
+        .rx             = GPIO_PC4_U1RX,
+        .tx             = GPIO_PC5_U1TX,
+        .mask_rx        = (GPIO_PIN_4),
+        .mask_tx        = (GPIO_PIN_5),
+        #endif
+    },
+    {
+        .uart_sysctl    = SYSCTL_PERIPH_UART2,
+        .uart_base      = UART2_BASE,
+        .uart_irq_chan  = UART2_IRQn,
+        .uart_im_r      = &UART2_IM_R,
+
+        .gpio_sysctl = SYSCTL_PERIPH_GPIOD,
+        .gpio_port   = GPIO_PORTD_BASE,
+        .rx             = GPIO_PD6_U2RX,
+        .tx             = GPIO_PD7_U2TX,
+        .mask_rx        = (GPIO_PIN_6),
+        .mask_tx        = (GPIO_PIN_7),
+    },
+};
+
+#define VALID_UART_CONFIGS  (sizeof(ucf) / sizeof(ucf[0]))
+
+#define UART_ASSERT assert(uart < UART_NUMOF); \
+                    assert(UART_NUMOF <= VALID_UART_CONFIGS);\
+                    assert(uart_config[uart].uart_num < VALID_UART_CONFIGS);
 
 /**
  * Configuring the UART console
  */
 int uart_init(uart_t uart, uint32_t baudrate, uart_rx_cb_t rx_cb, void *arg)
 {
     /* Check the arguments */
-    assert(uart == 0);
+    UART_ASSERT;
+
+    unsigned int num = uart_config[uart].uart_num;
+
     /* Check to make sure the UART peripheral is present */
-    if(!ROM_SysCtlPeripheralPresent(SYSCTL_PERIPH_UART0)){
+    if(!ROM_SysCtlPeripheralPresent(ucf[num].uart_sysctl)){
         return UART_NODEV;
     }
 
-    int res = init_base(uart, baudrate);
-    if(res != UART_OK){
-        return res;
+    ROM_SysCtlPeripheralEnable(ucf[num].uart_sysctl);
+    ROM_SysCtlPeripheralEnable(ucf[num].gpio_sysctl);
+
+    /* Configure the Rx and Tx pins. If no callback function is defined,
+     * the UART should be initialised in Tx only mode.
+     */
+    if (rx_cb) {
+        ROM_GPIOPinConfigure(ucf[num].rx);
```

Comment:
I suggest to group this with the two `if (rx_cb) ...` blocks below to save some instructions.